lib/platform/client/proxy_http.js

```diff
@@ -1,164 +1,228 @@
+        },
+
+        makeRequest: function(url, message) {
+            // Add our original destination to to headers,
+            // as some proxy implementations would rather
+            // use this.
+            message.headers["X-ProxyDestination"] = url;
+            message.headers["Splunk-Client"] = "splunk-sdk-javascript/" + SDK_VERSION;
+            if(message.response_timeout != undefined){
+                message.headers["Response-Timeout"] = message.response_timeout;
+            }
+            
+            // Need to remove the hostname from the URL
+            var parsed = parseUri(url);
+            var prefixToRemove = "" + (parsed.protocol ? parsed.protocol : "") + "://" + parsed.authority;
+            url = url.replace(new RegExp(escape(prefixToRemove), "i"), "");
+            
+            // Now, we prepend the prefix
+            url = this.prefix + url;
+            var that = this;
+            var complete_response;
+            var params = {
+                url: url,
+                type: message.method,
+                headers: message.headers,
+                data: message.body || "",
+                timeout: message.timeout || 0,
+                dataType: "text",
+                success: function(data, error, res) {
+                    var response = {
+                        statusCode: res.status,
+                        headers: getHeaders(res.getAllResponseHeaders())
+                    };
+
+                    complete_response = that._buildResponse(error, response, data);
+                    return Promise.resolve(complete_response);
+                },
+                error: function(res, data, error) {
+                    // Format abort response
+                    if(res.status === 600){
+                        data = JSON.parse(res.responseText).statusCode;
+                        let response = JSON.parse(res.responseText);
+                        complete_response = that._buildResponse("abort",response,{});
+                        return Promise.reject(complete_response);
+                    }
+                    var response = {
+                        statusCode: res.status,
+                        headers: getHeaders(res.getAllResponseHeaders())
+                    };
+
+                    if (data === "abort") {
+                        response.statusCode = "abort";
+                        res.responseText = "{}";
+                    }
+                    var json = res.responseText;
+
+                    complete_response = that._buildResponse(error, response, json);
+                    
+                    return Promise.reject(complete_response);
+                }
+            };
+            
+            return $.ajax(params).then((xhr)=>{
+                return complete_response;
+            }).catch((err)=>{
+                return complete_response;
+            });
+        },
+
+        makeRequestAsync: function(url, message) {

```

Comment:
This appears to be identical to `makeRequest` above - can we remove it?

Reply:
or was this meant to be `makeRequestAsync: async function(url, message) {` in which case can we have this function call makeRequest?

Reply:
we had added this method for `conf` CRUD functionality and it was implemented using Promises. Currently we did not merge this with the updated makeRequest method as it would require some analysis and testing on the working conf CRUD feature. we thought to include it as a follow-up once Promise changes are merged.